src/Nethermind/Nethermind.Consensus/Processing/ParallelProcessing/MultiVersionMemory.cs

```diff
@@ -0,0 +1,194 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Nethermind.Core.Collections;
+using Nethermind.Core.Extensions;
+
+namespace Nethermind.Consensus.Processing.ParallelProcessing;
+
+public class MultiVersionMemory<TLocation, TLogger>(ushort txCount, ParallelTrace<TLogger> parallelTrace) where TLogger : struct, IIsTracing where TLocation : notnull
+{
+    private readonly record struct Value(ushort Incarnation, byte[] Bytes);
```

Comment:
Why `Value` and this structure in general captures `byte[]`?

Reply:
Read the paper ;)